lib/elixir/lib/exception.ex

```diff
@@ -683,6 +693,10 @@ defmodule UndefinedFunctionError do
     "      * " <> fun <> "/" <> Integer.to_string(arity) <> "\n"
   end
 
+  defp macro_with_same_mfa?(module, function, arity) do
+    {:"MACRO-#{function}", arity + 1} in module.module_info(:exports)
```

Comment:
What about checking if `__info__/1` is exported and then using `module.__info__(:macros)` instead? If there's an erlang module with a strangely named function this might behave strangely. WDYT? :)